### PR TITLE
[JENKINS-41795] Report event origins

### PR DESCRIPTION
**Pull request by stephenc**
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/27

See [JENKINS-41795](https://issues.jenkins-ci.org/browse/JENKINS-41795)

Reports the origin of events, the event logs will now look something like:

screen shot 2017-02-07 at 11 56 09

And the tooltip for a branch event will include the origin if known:

screen shot 2017-02-07 at 11 57 40

@reviewbybees

### CODE REVIEW

**Comment by KostyaSha** [comment]
Exposing debug information to end users looks like overkill. Who cares about this IPs?

File: src/main/java/jenkins/branch/BranchEventCause.java [hunk 1]
```diff
@@ -62,6 +64,10 @@ public Date getTimestamp() {
         return new Date(timestamp);
     }
 
+    public String getOrigin() {
```
Comment (jglick) [inline src/main/java/jenkins/branch/BranchEventCause.java]: `@CheckForNull`

File: src/main/resources/jenkins/branch/BranchEventCause/description.properties [hunk 2]
```diff
@@ -1 +1,2 @@
 blurb=Branch event at {0,time} on {0,date}
+origin=Origin: {0}
```
Comment (jglick) [inline src/main/resources/jenkins/branch/BranchEventCause/description.properties]: Will this show `Origin: ?` if the origin is unknown?

Comment (stephenc) [inline src/main/resources/jenkins/branch/BranchEventCause/description.properties]: yes

**Comment by stephenc** [comment]
@KostyaSha said:

> Exposing debug information to end users looks like overkill. Who cares about this IPs?

When the user is trying to determine why they have strange extra builds being triggered the IP address of the origin of the requests is vital.

There is no value in hiding the information from users as it will be in the logs that anyone with Item/READ can access anyway, and we cannot hide the origin in the logs from non-admins (plus the origin is expected to normally be the SCM system - given that the SCM links will be pointing to that system, its IP address is not exactly top secret)

The origin is only available as a tool-tip so not exactly in your face either

File: src/main/java/jenkins/branch/BranchEventCause.java (hunk 1 above)
Comment (stephenc) [inline src/main/java/jenkins/branch/BranchEventCause.java]: `@NonNull` because the event will substitute `?`

**Pull request merged**